### PR TITLE
[Commands] Add LinuxMainGenerator

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
This adds a new option "--generate-linuxmain" to 'swift test' to
generate test entries that can be used for XCTest on Linux.  Since only
macOS supports automatic test discovery, this option will only work on
macOS. Using it on Linux will result in a "null" generation.

<rdar://problem/37259907> Use macOS's test disovery to generate LinuxMain.swift

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

File: Sources/Commands/SwiftTestTool.swift [hunk 1]
```diff
@@ -136,8 +146,17 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
                 print(test.specifier)
             }
 
+        case .generateLinuxMain:
+          #if os(Linux)
```
Comment (masters3d) [inline Sources/Commands/SwiftTestTool.swift]: should this also be ` #if !os(macOS)`?

Comment (aciidb0mb3r) [reply Sources/Commands/SwiftTestTool.swift]: Yeah, that will be more accurate. Thanks!

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Pull request merged**